### PR TITLE
Provisioning AWS Databricks E2 with a Hub & Spoke firewall

### PR DESCRIPTION
**Pull request by kivanc-databricks**
Provisioning AWS Databricks E2 with a Hub & Spoke firewall for data exfiltration protection

### CODE REVIEW

File: docs/guides/aws-e2-firewall-hub-and-spoke.md [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+  count                   = length(local.hub_tgw_private_subnets_cidr)
+  cidr_block              = element(local.hub_tgw_private_subnets_cidr, count.index)
+  availability_zone       = element(local.availability_zones, count.index)
+  map_public_ip_on_launch = false
+  tags = merge(var.tags, {
+    Name = "${local.prefix}-hub-tgw-private-${element(local.availability_zones, count.index)}"
+  })
+}
+
+/* NAT Public subnet */
```
Comment (nfx) [inline docs/guides/aws-e2-firewall-hub-and-spoke.md:271]: general feeling about this doc - it's a **huge intimidating blob of config**. don't assume that people would be reading blog post to see what things are what.

could you split the guide into a bit more sections with text in between different config blocks, probably just copy-paste those from the blog post :) this way it'll be easier to read. try to make a paragraph of text for every 30-50 lines of config.

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by nfx** [comment]
@kivanc-databricks thank you for testing this whole thing out and making an effort to document it!

File: docs/guides/aws-e2-firewall-hub-and-spoke.md (hunk 1 above)
Comment (nfx) [reply docs/guides/aws-e2-firewall-hub-and-spoke.md:271]: Probably re-using the source of diagram in the blogpost to create section-specific diagrams would also make this way more friendly and readable.

Comment (kivanc-databricks) [reply docs/guides/aws-e2-firewall-hub-and-spoke.md:308]: Added some diagrams

**Review by nfx — APPROVED** [review]

**Pull request merged**